<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>767</width>
    <height>463</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Шифр Альберти</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_9">
    <item>
     <layout class="QHBoxLayout" name="upBlockLayout">
      <item>
       <layout class="QVBoxLayout" name="leftBlockLayout">
        <property name="rightMargin">
         <number>6</number>
        </property>
        <item>
         <layout class="QVBoxLayout" name="keyLayout">
          <item>
           <widget class="QLabel" name="key_label">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="maximumSize">
             <size>
              <width>69</width>
              <height>44</height>
             </size>
            </property>
            <property name="font">
             <font>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="cursor">
             <cursorShape>WhatsThisCursor</cursorShape>
            </property>
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Ключ на внутреннем диске по часовой стрелке; начальная позиция - напротив буквы &amp;quot;А&amp;quot; внешнего диска.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="statusTip">
             <string/>
            </property>
            <property name="whatsThis">
             <string/>
            </property>
            <property name="text">
             <string>Ключ</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="keyEditLayout">
            <item>
             <widget class="QLineEdit" name="key_edit">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>260</width>
                <height>30</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>16777215</width>
                <height>25</height>
               </size>
              </property>
              <property name="font">
               <font>
                <pointsize>12</pointsize>
               </font>
              </property>
              <property name="cursor">
               <cursorShape>IBeamCursor</cursorShape>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="keyGenerateButton">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>100</width>
                <height>30</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>16777215</width>
                <height>23</height>
               </size>
              </property>
              <property name="font">
               <font>
                <pointsize>12</pointsize>
               </font>
              </property>
              <property name="text">
               <string>Сгенерировать!</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="openTextLayout">
          <item>
           <widget class="QLabel" name="openText_label">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="maximumSize">
             <size>
              <width>125</width>
              <height>18</height>
             </size>
            </property>
            <property name="font">
             <font>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="cursor">
             <cursorShape>WhatsThisCursor</cursorShape>
            </property>
            <property name="toolTip">
             <string>Здесь выводится открытый текст после расшифровки</string>
            </property>
            <property name="text">
             <string>Открытый текст</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPlainTextEdit" name="openText_edit">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="font">
             <font>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="cursor" stdset="0">
             <cursorShape>IBeamCursor</cursorShape>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="rightBlockLayout">
        <item>
         <widget class="QLabel" name="cipherText_label">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>99</width>
            <height>18</height>
           </size>
          </property>
          <property name="baseSize">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
          <property name="font">
           <font>
            <pointsize>12</pointsize>
           </font>
          </property>
          <property name="cursor">
           <cursorShape>WhatsThisCursor</cursorShape>
          </property>
          <property name="toolTip">
           <string>Здесь выводится шифротекст после зашифровки</string>
          </property>
          <property name="text">
           <string>Шифротекст</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPlainTextEdit" name="cipherText_edit">
          <property name="cursor" stdset="0">
           <cursorShape>IBeamCursor</cursorShape>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="downBlockLayout" stretch="0,0">
      <property name="topMargin">
       <number>6</number>
      </property>
      <item>
       <layout class="QHBoxLayout" name="settingsLayout" stretch="0,0,0,0">
        <item>
         <widget class="QGroupBox" name="actionBox">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>170</width>
            <height>85</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>170</width>
            <height>85</height>
           </size>
          </property>
          <property name="cursor">
           <cursorShape>ArrowCursor</cursorShape>
          </property>
          <property name="title">
           <string>Действие:</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
          </property>
          <widget class="QWidget" name="verticalLayoutWidget_2">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>20</y>
             <width>161</width>
             <height>54</height>
            </rect>
           </property>
           <layout class="QVBoxLayout" name="actionLayout" stretch="1,1">
            <property name="leftMargin">
             <number>5</number>
            </property>
            <item>
             <widget class="QRadioButton" name="encryptRadioButton">
              <property name="font">
               <font>
                <pointsize>12</pointsize>
               </font>
              </property>
              <property name="text">
               <string>Шифрование</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="decryptRadioButton">
              <property name="font">
               <font>
                <pointsize>12</pointsize>
               </font>
              </property>
              <property name="text">
               <string>Расшифрование</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>60</width>
            <height>85</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>60</width>
            <height>85</height>
           </size>
          </property>
          <property name="title">
           <string>Алфавит</string>
          </property>
          <widget class="QRadioButton" name="radioButton">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>20</y>
             <width>51</width>
             <height>17</height>
            </rect>
           </property>
           <property name="text">
            <string>RU</string>
           </property>
          </widget>
          <widget class="QRadioButton" name="radioButton_6">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>50</y>
             <width>51</width>
             <height>17</height>
            </rect>
           </property>
           <property name="text">
            <string>ENG</string>
           </property>
          </widget>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="modeBox">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>70</width>
            <height>87</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>70</width>
            <height>87</height>
           </size>
          </property>
          <property name="cursor">
           <cursorShape>WhatsThisCursor</cursorShape>
          </property>
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;1) Первая буква сообщения совмещается сама с собой&lt;/p&gt;&lt;p&gt;2) Первая буква сообщения совмещается с первой буквой алфавита&lt;/p&gt;&lt;p&gt;3) Данная индикаторная буква выставляется напротив первой буквы алфавита&lt;/p&gt;&lt;p&gt;4) Блочный: текст шифруется блоками данного размера&lt;/p&gt;&lt;p&gt;5) По паролю: буквы данного пароля поочередно совмещаются с первой буквой алфавита&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="title">
           <string>Режим:</string>
          </property>
          <widget class="QComboBox" name="modeComboBox">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>20</y>
             <width>51</width>
             <height>22</height>
            </rect>
           </property>
           <item>
            <property name="text">
             <string>1</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>2</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>3</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>4</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>5</string>
            </property>
           </item>
          </widget>
         </widget>
        </item>
        <item>
         <layout class="QGridLayout" name="gridLayout">
          <item row="1" column="0">
           <layout class="QVBoxLayout" name="blockSizeLayout">
            <item>
             <widget class="QLabel" name="blockSize_label">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="text">
               <string>Размер блока</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="blockSize_edit">
              <property name="enabled">
               <bool>false</bool>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="1" column="1">
           <layout class="QVBoxLayout" name="verticalLayout_2">
            <item>
             <widget class="QLabel" name="chosenLetter_label">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="text">
               <string>Индикаторный символ</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="chosenLetter_edit">
              <property name="enabled">
               <bool>false</bool>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="0" column="0" colspan="2">
           <layout class="QVBoxLayout" name="keyPhraseLayout">
            <item>
             <widget class="QLabel" name="keyPhrase_label">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="text">
               <string>Ключевая фраза</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="keyPhrase_edit">
              <property name="enabled">
               <bool>false</bool>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QPushButton" name="startButton">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Maximum" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>200</width>
          <height>80</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>12</pointsize>
         </font>
        </property>
        <property name="cursor">
         <cursorShape>PointingHandCursor</cursorShape>
        </property>
        <property name="text">
         <string>Старт!</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>767</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>Меню</string>
    </property>
    <addaction name="Import"/>
    <addaction name="Export"/>
    <addaction name="separator"/>
    <addaction name="Exit"/>
   </widget>
   <addaction name="menu"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="Export">
   <property name="text">
    <string>Экспорт в файл...</string>
   </property>
   <property name="statusTip">
    <string>Записать алфавит и шифротекст в файл</string>
   </property>
  </action>
  <action name="Import">
   <property name="text">
    <string>Импорт из файла...</string>
   </property>
   <property name="statusTip">
    <string>Считать входные данные из файла</string>
   </property>
   <property name="whatsThis">
    <string/>
   </property>
  </action>
  <action name="Exit">
   <property name="text">
    <string>Выход</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
